lib/check.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, lib, haskellLib }:
+drv:
+
+let
+  component = drv.config;
+
+# This derivation can be used to execute test component.
+# The $out of the derivation is a file containing the resulting
+# stdout output.
+in stdenv.mkDerivation ({
+  name = (drv.name + "-run");
+
+  src = drv;
```

Comment:
I'm not sure whether this is going to work for cardano-wallet. Lots of tests require data files from the package source tree. So perhaps `src` should be passed in as an argument from the main builder. Then the check phase would run the executable from within the `$src` directory.

Reply:
Do they use `src` env variable or current dir to find it?  Do they use the `Paths_` modules?

Reply:
Sticking test data in `data:` and addressing it via `Paths_` module is probably not great in a lot of cases (as it means it would be in the closure for the `library` and `exes` as well).  I'll fix it up do src is the same as drv src.